v3/src/components/web-view/use-data-interactive-controller.tsx

```diff
@@ -0,0 +1,74 @@
+import { useToast } from "@chakra-ui/react"
+import iframePhone from "iframe-phone"
+import React, { useEffect } from "react"
+import { DEBUG_PLUGINS, debugLog } from "../../lib/debug"
+import { ITileModel } from "../../models/tiles/tile-model"
+import { isWebViewModel } from "./web-view-model"
+
+function extractOrigin(url?: string) {
+  if (!url) return
+  return new URL(url).origin
+}
+
+export function useDataInteractiveController(iframeRef: React.MutableRefObject<HTMLIFrameElement|null>, tile?: ITileModel) {
```

Comment:
I think you can use:
```suggestion
export function useDataInteractiveController(iframeRef: React.RefObject<HTMLIFrameElement>, tile?: ITileModel) {
```
which has the advantage that it makes the line <120. 😉